app/Http/Controllers/AdvertController.php

```diff
@@ -41,7 +41,7 @@ public function store(CreateSubscriptionRequest $request): JsonResponse
      * @param Advert $advert
      * @return JsonResponse
      */
-    public function destroy(Request $request, Advert $advert): JsonResponse
+    public function destroy(Advert $advert, Request $request): JsonResponse
     {
         return response()->json($this->subscribeService->removeSubscription($request->user(), $advert));
```

Comment:
Actually you can use auth()->user() wthout extra $request

Reply:
I think using global helpers is less flexible than explicit dependency injection